console-subscriber/src/init.rs

```diff
@@ -0,0 +1,83 @@
+use crate::TasksLayer;
+use std::thread;
+use tokio::runtime;
+use tracing_core::{subscriber::Subscriber, Dispatch};
+use tracing_subscriber::{fmt, layer::Layered, prelude::*, EnvFilter, Layer, Registry};
+
+type ConsoleSubscriberLayer =
+    Layered<TasksLayer, Layered<EnvFilter, Layered<fmt::Layer<Registry>, Registry>>>;
+
+/// Starts the console subscriber server on its own thread
+///
+/// This function represents the easiest way to get started using
+/// tokio-console.
+///
+/// ## Configuration
+///
+/// Tokio console subscriber is configured with sensible defaults for most
+/// use cases. If you need to tune these parameters, several environmental
+/// configuration variables are available:
+///
+/// * `TOKIO_CONSOLE_RETENTION_SECS`: The number of seconds to accumulate
+///   completed tracing data. Default: 60s.
+/// * `TOKIO_CONSOLE_BIND`: a HOST:PORT description, such as
+///   localhost:1234 or similar. Default: 127.0.0.1:6669
+/// * `TOKIO_CONSOLE_PUBLISH_INTERVAL_MS`: The number of milliseconds to wait
+///   between sending updates to the console. Default: 1000ms (1s)
+/// * `RUST_LOG`: configure the tracing filter. Default: `tokio=trace`,
+///   and any additional filtering directives will be appended to this
+///   default. See [`EnvFilter`] for further information on the format
+///   for this variable.
+///
+/// ## Adding additional layers
+///
+/// To add an additional [`Layer`], see [`init_with_layer`]
+pub fn init() {
+    init_inner(|console_subscriber| console_subscriber)
+}
+
+/// Starts the console subscriber server with an additional subscriber
+/// layer, for example a log layer. This interface can be combined
+/// with any of the environment configuration documented at [`init`].
+pub fn init_with_layer<AdditionalLayer>(additional_layer: AdditionalLayer)
+where
+    AdditionalLayer: Layer<ConsoleSubscriberLayer> + Send + Sync + 'static,
```

Comment:
an alternative API design is to have a function that spawns the background thread, and just returns a `Registry` with the `EnvFilter` and `TasksLayer` layers already added. that way, the user can call any methods on that `Registry` to add additional layers, and set it as the default subscriber themselves. the `init` function could just call that and then immediately call `init()` on the returned registry. 

i feel like this design could be more intuitive than taking a layer as an argument, but it might be _slightly_ wordier to use...i could be convinced either way.

Reply:
That does feel more flexible. What if instead of init_with_layer it was `build() -> impl Layer + Into<Dispatch>`, with `#[must_use]`.

```
console_subscriber::build().with(…).init()
console_subscriber::init(); // shorthand for console_subscriber::build().init();
```

Reply:
addressed in bb2c2dbef5d03e6d23cca0161b03e1f5471b8119 — build and init